A. Commit message:
geneve: Suppress list corruption splat in geneve_destroy_tunnels()

B. Change summary:
Removed the for_each_netdev() iteration in geneve_destroy_tunnels() to prevent triggering geneve_dellink() twice for the same device, which could cause list corruption when CONFIG_DEBUG_LIST is enabled. Delegated that part to default_device_exit_batch().

C. Compatibility Risk:
Low

D. Fixed Code:
```diff
diff --git a/drivers/net/geneve.c b/drivers/net/geneve.c
index a1f674539965d..dbb3960126ee7 100644
--- a/drivers/net/geneve.c
+++ b/drivers/net/geneve.c
@@ -1902,14 +1902,7 @@ static void geneve_destroy_tunnels(struct net *net, struct list_head *head)
 {
 	struct geneve_net *gn = net_generic(net, geneve_net_id);
 	struct geneve_dev *geneve, *next;
-	struct net_device *dev, *aux;
 
-	/* gather any geneve devices that were moved into this ns */
-	for_each_netdev_safe(net, dev, aux)
-		if (dev->rtnl_link_ops == &geneve_link_ops)
-			geneve_dellink(dev, head);
-
-	/* now gather any other geneve devices that were created in this ns */
 	list_for_each_entry_safe(geneve, next, &gn->geneve_list, next)
 		geneve_dellink(geneve->dev, head);
 }
```